barretenberg/cpp/src/barretenberg/proof_system/circuit_builder/toy_avm/toy_avm_circuit_builder.hpp

```diff
@@ -66,12 +72,62 @@ template <typename Flavor> class ToyAVMCircuitBuilder {
             polys.permutation_set_column_3[i] = wires[2][i];
             polys.permutation_set_column_4[i] = wires[3][i];
             polys.self_permutation_column[i] = wires[4][i];
+
             // By default the permutation is over all rows where we place data
             polys.enable_tuple_set_permutation[i] = 1;
             // The same column permutation alternates between even and odd values
             polys.enable_single_column_permutation[i] = 1;
             polys.enable_first_set_permutation[i] = i & 1;
             polys.enable_second_set_permutation[i] = 1 - (i & 1);
+
+            // Lookup-based range constraint related values
+
+            // Store the value
+            polys.range_constrained_column[i] = wires[5][i];
+            // Make range constrained
+            polys.lookup_is_range_constrained[i] = 1;
+            uint256_t constrained_value = wires[5][i];
+            // if the value is correct, update the appropriate counter
+            if (constrained_value < 256) {
+                polys.lookup_range_constraint_read_count[static_cast<size_t>(constrained_value.data[0])] =
+                    polys.lookup_range_constraint_read_count[static_cast<size_t>(constrained_value.data[0])] + 1;
+            }
+
+            // Copy xor values
+            polys.lookup_xor_argument_1[i] = wires[6][i];
+            polys.lookup_xor_argument_2[i] = wires[7][i];
+            polys.lookup_xor_result[i] = wires[8][i];
+            polys.lookup_xor_accumulated_argument_1[i] = wires[9][i];
```

Comment:
what do `lookup_xor_accumulated_argument_1` and `lookup_xor_accumulated_argument_2` represent?

How many columns are we adding per distinct lookup?

Reply:
We are trying to prove that `lookup_xor_accumulated_argument_1 ^ lookup_xor_accumulated_argument_2 = lookup_xor_accumulated_result`. However they are 8-bit, so we decompose them. The top 4 bits of lookup_xor_accumulated_argument_1 will be in lookup_xor_argument_1.

For this particular case we added 6 columns for arguments. 2 3-element tuples.